File: packages/server/PostService/internal/domain/models/post.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Post struct {
	User        primitive.ObjectID   `json:"user" bson:"user"`
	Title       string               `json:"title" bson:"title"`
	Content     string               `json:"content" bson:"content"`
	CreatedAt   time.Time            `json:"createdAt" bson:"createdAt"`
	Likes       int                  `json:"likes" bson:"likes"`
	Dislikes    int                  `json:"dislikes" bson:"dislikes"`
	HeaderImage string               `json:"headerImage" bson:"headerImage"`
	Comments    []primitive.ObjectID `json:"comments" bson:"comments"`
	Tags        []string             `json:"tags" bson:"tags"`
}

File: packages/server/UserService/Models/User/Post.cs
using System.Text.Json.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using UserService.Dto;

namespace UserService.Models.User;
public class Post
{
   [BsonId]
   [BsonRepresentation(BsonType.ObjectId)]
   [JsonPropertyName("_id")]
   public string Id { get; set; }
   [BsonElement("title")]
   public string Title { get; set; }
   
   [BsonElement("user")] 
   [BsonRepresentation(BsonType.ObjectId)]
   public ObjectId UserId { get; set; }
   
   [BsonElement("content")]
   public string Content { get; set; }
   [BsonElement("headerImage")]
   public string HeaderImage { get; set; }
   [BsonElement("createdAt")]
   public DateTime CreatedAt { get; set; }
   [BsonElement("likes")]
   public int Likes { get; set; }
   [BsonElement("dislikes")]
   public int Dislikes { get; set; }
   
   [BsonElement("saved")]
   public int Saved { get; set; }
   
   [BsonElement("tags")] 
   public List<string> Tags { get; set; }

   [BsonElement("comments")]
   [BsonRepresentation(BsonType.ObjectId)]
   public List<string> Comments { get; set; }
   
   [BsonElement("reports")]
   [BsonRepresentation(BsonType.ObjectId)]
   public List<string> Reports { get; set; }
}